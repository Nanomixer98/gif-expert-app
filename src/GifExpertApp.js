import React, { useState } from 'react';
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

export const GifExpertApp = () => {

    // const categories = ['Pokemon', 'Half Life', 'Halo'];

    const [categories, setCategories] = useState(['Halo']);

    // const handleAdd = () => {
    //     // setCategories([...categories, 'COD']);
    //     setCategories(cats => [...cats, 'COD']);
    // }



    return (
    <>
        <h2>GifExpertApp</h2>
        <AddCategory setCategories={ setCategories }/>
        <hr/>

        {/* <button onClick={handleAdd}>Agregar</button> */}
        <ol>
            {
                categories.map((category) => ( 
                    <GifGrid 
                        key={ category }
                        category={ category }
                    />
                ))
            }
        </ol>
    </>);
  }