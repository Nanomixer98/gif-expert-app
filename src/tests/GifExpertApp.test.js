import { GifExpertApp } from "../GifExpertApp";
import "@testing-library/jest-dom";
import React from 'react';
import { shallow } from 'enzyme';

describe('Prueba de renderizado index', () =>{

    let wrapper = shallow(<GifExpertApp/>);
    test('Debe mostrar correctamente <GifExpertApp />', () => { 
        expect(wrapper).toMatchSnapshot();
    });

});