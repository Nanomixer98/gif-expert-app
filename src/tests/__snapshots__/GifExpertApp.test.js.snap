// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba de renderizado index Debe mostrar correctamente <GifExpertApp /> 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Halo"
      key="Halo"
    />
  </ol>
</Fragment>
`;
